espnet2/tasks/vc.py

```diff
@@ -0,0 +1,437 @@
+            idim = input_feats_extract.output_size()
+        else:
+            # Give features from data-loader
+            args.input_feats_extract = None
+            args.input_feats_extract_conf = None
+            input_feats_extract = None
+            idim = args.idim
+
+        # 2. Normalization layer
+        if args.normalize is not None:
+            normalize_class = normalize_choices.get_class(args.normalize)
+            normalize = normalize_class(**args.normalize_conf)
+        else:
+            normalize = None
+        if args.input_normalize is not None:
+            input_normalize_class = input_normalize_choices.get_class(
+                args.input_normalize
+            )
+            input_normalize = input_normalize_class(**args.input_normalize_conf)
+        else:
+            input_normalize = None
+
+        # 3. VC
+        vc_class = vc_choices.get_class(args.vc)
+        vc = vc_class(idim=idim, odim=odim, **args.vc_conf)
+
+        # 4. Extra components
+        pitch_extract = None
+        energy_extract = None
+        pitch_normalize = None
+        energy_normalize = None
+        if getattr(args, "pitch_extract", None) is not None:
+            pitch_extract_class = pitch_extractor_choices.get_class(args.pitch_extract)
+            if args.pitch_extract_conf.get("reduction_factor", None) is not None:
+                assert args.pitch_extract_conf.get(
+                    "reduction_factor", None
+                ) == args.vc_conf.get("reduction_factor", 1)
+            else:
+                args.pitch_extract_conf["reduction_factor"] = args.vc_conf.get(
+                    "reduction_factor", 1
+                )
+            pitch_extract = pitch_extract_class(**args.pitch_extract_conf)
+        if getattr(args, "energy_extract", None) is not None:
+            if args.energy_extract_conf.get("reduction_factor", None) is not None:
+                assert args.energy_extract_conf.get(
+                    "reduction_factor", None
+                ) == args.vc_conf.get("reduction_factor", 1)
+            else:
+                args.energy_extract_conf["reduction_factor"] = args.vc_conf.get(
+                    "reduction_factor", 1
+                )
+            energy_extract_class = energy_extractor_choices.get_class(
+                args.energy_extract
+            )
+            energy_extract = energy_extract_class(**args.energy_extract_conf)
+        if getattr(args, "pitch_normalize", None) is not None:
+            pitch_normalize_class = pitch_normalize_choices.get_class(
+                args.pitch_normalize
+            )
+            pitch_normalize = pitch_normalize_class(**args.pitch_normalize_conf)
+        if getattr(args, "energy_normalize", None) is not None:
+            energy_normalize_class = energy_normalize_choices.get_class(
+                args.energy_normalize
+            )
+            energy_normalize = energy_normalize_class(**args.energy_normalize_conf)
```

Comment:
If we use these extractors, it should not be shared between src trg spks as well as feats_extract.